pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,384 @@
+
+        if self.use_lottery_ticket_hypothesis:
+            # make a copy of copy of orginal weights.
+            self._parameters_to_prune = [(deepcopy(m), n) for m, n in self.parameters_to_prune]
+
+    def on_epoch_end(self, trainer, pl_module):
+        self.apply_pruning(trainer, pl_module)
+
+        if self.make_pruning_permanent:
+            self._make_pruning_permanent()
+
+    @staticmethod
+    def _sanitize_parameters_to_prune(p):
+        """
+        Check the provide element is a pair with:
+            * nn.Module
+            * str
+
+        Example::
+
+            parameters_to_prune = [
+                (model.mlp_1, "weight"),
+                (model.mlp_2, "weight")
+            ]
+        """
+        return len(p) == 2 and isinstance(p[0], nn.Module) and isinstance(p[1], str)
+
+    @staticmethod
+    def sanitize_parameters_to_prune(
+        pl_module: LightningModule,
+        parameters_to_prune: Optional[List[Tuple[nn.Module, str]]],
+        parameters: List[str] = ["weight"]
+    ) -> List:
+        """
+        This function is responsible to check provided `parameters_to_prune` and `parameters`.
+        If parameters_to_prune is None, parameters_to_prune will be generated from all parameters of the model.
+        """
+
+        is_parameters_to_prune_none = parameters_to_prune is None
+        current_modules = [
+            m for m in pl_module.modules()
+            if not isinstance(m, (LightningModule, ModuleDict, ModuleList))
+        ]
+
+        if is_parameters_to_prune_none:
+            parameters_to_prune = []
+            for p in parameters:
+                for m in current_modules:
+                    param = getattr(m, p, None)
+                    if param is not None:
+                        parameters_to_prune.append((m, p))
+
+        if isinstance(parameters_to_prune, (tuple, list)) \
+           and len(parameters_to_prune) > 0 and not is_parameters_to_prune_none:
+
+            if all(
+                isinstance(p, (list, tuple)) and ModelPruning._sanitize_parameters_to_prune(p)
+                for p in parameters_to_prune
+            ):
+
+                missing_modules = []
+                missing_parameters = []
+
+                for module, param_name in parameters_to_prune:
+                    if module not in current_modules:
+                        missing_modules.append(module)
+                        continue
+
+                    parameter = getattr(module, param_name)
+
+                    if parameter is None:
+                        missing_parameters.append(parameter)
+
+                if len(missing_modules) > 0 or len(missing_parameters) > 0:
+                    raise MisconfigurationException(
+                        "Ths provided parameters_to_tune doesn't exist in the model. "
+                        f"Found mismatching modules: {missing_modules} and missing_parameters: {missing_parameters}"
```

Comment:
```suggestion
                        "Ths provided parameters_to_tune doesn't exist in the model."
                        f" Found mismatching modules: {missing_modules} and missing_parameters: {missing_parameters}"
```
just tobe the same in all codebase